wand-watcher.lic

```diff
@@ -0,0 +1,179 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#wand-watcher
+=end
+
+no_kill_all
+no_pause_all
+
+custom_require.call(%w[common common-items])
+
+class WandWatcher
+  include DRC
+  include DRCI
+
+  def initialize
+    settings = get_settings
+    @wand_list = settings.wands
+    @queued_wands = []
+    @no_use_scripts = settings.wand_watcher_no_use_scripts || []
+    @no_use_rooms = settings.wand_watcher_no_use_rooms || []
+    @startup_delay = settings.wand_watcher_startup_delay || 10
+    @passive_delay = settings.wand_watcher_passive_delay || 60
+    @activation_failure_messages = [/^The .* remains inert/]
+    @activation_blocked_messages = [/^Something in the area is interfering/,
+                                    /^This is not a good place for that/,
+                                    /^You shouldn't disrupt the area/,
+                                    /^You really shouldn't be loitering in here/]
+    UserVars.wand_watcher_timers = {} unless UserVars.wand_watcher_timers
+
+    # Don't allow users to make foolish mistakes, always prevent use during go2/burgle
+    @no_use_scripts << 'go2'
+    @no_use_scripts << 'burgle'
+
+    # Rebuild UserVars based on what wands are listed in the yaml, this:
+    # - preserves timers for existing wands
+    # - removes timers for wands no longer in the yaml
+    # - sets any new wand next use timers to now
+    temp_timers = {}
+    @wand_list.each do |key,item|
+      if UserVars.wand_watcher_timers[key]
+        temp_timers[key] = UserVars.wand_watcher_timers[key]
+      else
+        temp_timers[key] = Time.now
+      end
+    end
+    UserVars.wand_watcher_timers = temp_timers
+    
+    # Pause for the startup delay so you can control spammyness of autostart scripts
+    pause @startup_delay
+
+    # Run passively in a loop
+    passive_run
+    #TODO: run once option
+  end
+
+  def passive_run
+    loop do
+      if @wand_list == nil || @wand_list.empty?
+        # @wand_list = nil when no wands configured in yaml
+        # @wand_list = empty when all wands are deleted due to problems finding wands
+        message "No wands to check.  Ending script.  Double check your settings."
+        exit
+      end
+      check_timers
+      pause @passive_delay
+    end
+  end
+
+  def check_timers  
+    # If any no_use_scripts are running, or you are in any of the no_use_rooms, 
+    # loop every 10 seconds checking until not
+    loop do
+      break if !(@no_use_scripts.any? { |name| Script.running?(name) }) && !( @no_use_rooms.any? { |room| room === DRRoom.title.to_s()[2..-3] || room == Room.current.id} )
```

Comment:
I may or may not have been skulking in the shadows when I was forced out of my hiding spot to use a wand, subsequently spooking my mark. 😁 

```suggestion
      break if !invisible? && !hidden? && !(@no_use_scripts.any? { |name| Script.running?(name) }) && !( @no_use_rooms.any? { |room| room === DRRoom.title.to_s()[2..-3] || room == Room.current.id} )
```

Reply:
I'm not convinced on this one tbh.  This could create a race condition where you never end up using it in combat if you're using stealth and the timers interact badly.  I feel like the better option is remembering to kill the script when you don't want it taking over for you.

Reply:
Good point. I don't train stealth in combat and hadn't thought of that scenario. I can easily kill this or other scripts I don't want interfering with something else I'm doing. 👍

Reply:
(reposting reply as KatoakDR — trying to keep my GitHub profile on theme!)

Good point. I don't train stealth in combat and hadn't thought of that scenario. I can easily kill this or other scripts I don't want interfering with something else I'm doing. 👍